src/abs.ott

```diff
@@ -255,6 +321,70 @@ defn
   ---------------------------- :: var
   F1 ... Fn , s |- x ~> s |- t
 
+  (-) z = z'
+  ---------------------------------------------------- :: neg
+  F1 ... Fn , s |- - z ~> s |- z'
+
+  (!) b = b'
+  ---------------------------------------------------- :: not
+  F1 ... Fn , s |- ! b ~> s |- b'
```

Comment:
Why have this premise at all? You can just put `(!) b` in place of `b'`. Same for `z'` in rule above.